lib/ansible/modules/windows/win_firewall_rule.ps1

```diff
@@ -156,7 +156,7 @@ try {
     # the default for enabled in module description is "true", but the actual COM object defaults to "false" when created
     if ($null -ne $enabled) { $new_rule.Enabled = $enabled } else { $new_rule.Enabled = $true }
     if ($null -ne $description) { $new_rule.Description = $description }
-    if ($null -ne $program -and $program -ne "any") { $new_rule.ApplicationName = $program }
+    if ($null -ne $program -and $program -ne "any") { $new_rule.ApplicationName =  [System.Environment]::ExpandEnvironmentVariables($program) }
```

Comment:
Instead of always setting the value as the expanded environment var we should probably just check the expanded version, or someone get the raw string back on our idempotency checks. This still allows people to specify the program using an expand string and we check it properly.

Also having a test for this scenario added to our integration tests would be great.

Reply:
The only place I see that we can expand the path is on line 185 inside the `$userPassedArguments` array

Reply:
I haven't looked at the code so no idea what the COM function returns on our checks. If that auto expands the value then we should do that on our checks. If the value is actually stored as an expanded string, i.e. the GUI doesn't show the expand string value, then maybe what you have here is correct. Really just needs some more investigation.

Reply:
I checked it out before the PR, it saves it as expanded that was the source of the issue

Reply:
This change fail, only expanding in 159 seems to do the trick.

our other option is adding all kinds of IFs in the code to expand properties where they relate to applicationname